python/cinn/compiler/compute_code_generator.py

```diff
@@ -0,0 +1,190 @@
+
+        Args:
+            node(ast.FunctionDef): The ast FunctionDef Node
+        """
+        arg_names = self.visit(node.args)
+
+        assert len(node.args.defaults) == 0, "Not support default args"
+
+        # 1. Construct args of function
+        llir_args = []
+        for i, arg_name in enumerate(arg_names):
+            if hasattr(self.inputs_signature[i], "dtype"):
+                llir_value = ir._Buffer_.make(
+                    "_" + arg_name, self.inputs_signature[i].dtype
+                )
+                llir_args.append(
+                    ir.Argument(llir_value, ir.Argument.IO.kUnknown)
+                )
+                tensor_shape = [
+                    ir.Expr(dim) for dim in self.inputs_signature[i].shape
+                ]
+                llir_value = ir._Tensor_.make(
+                    arg_name,
+                    self.inputs_signature[i].dtype,
+                    tensor_shape,
+                    tensor_shape,
+                )
+            else:
+                llir_value = ir.Var(arg_name)
+                llir_args.append(ir.Argument(llir_value))
+                llir_value = ir.Expr(llir_value)
+            self.set_value(arg_name, llir_value)
+
+        # 2. Construct body of function
+        stmts = self.visit_compound_statement(node.body)
+        body = ir.Block.make(stmts)
+
+        # 3. Construct LoweredFunc
+        self.cinn_llir_func = ir.LoweredFunc.make(
+            self.function_name, llir_args, body
+        )
+
+    def visit_compound_statement(self, stmts):
+        cinn_stmts = []
+        for stmt in stmts:
+            cinn_stmt = self.visit(stmt)
+            assert cinn_stmt is not None, f"Unsupport parse:\n{stmt}"
+            if cinn_stmt is None:
+                continue
+            cinn_stmts.append(self.visit(stmt))
+        return cinn_stmts
+
+    def visit_arguments(self, node):
+        arg_names = []
+        for arg in node.args:
+            arg_names += [self.visit(arg)]
+        return arg_names
+
+    def visit_arg(self, node):
+        ast.NodeVisitor.generic_visit(self, node)
+        return node.arg
+
+    def visit_For(self, node) -> ir.Expr:
+        """
+        parse CINN Low Level IR For.
+
+        Args:
+            node(ast.For): The ast For node
+
+        Returns:
+            ir.Expr, Points to the Expr of ir::ExprNode<For>
+        """
+        iter_args = [self.visit(arg) for arg in node.iter.args]
+        assert (
+            len(iter_args) <= 2
+        ), "CINN Low Level IR does not support setting the range step"
+        ast_min = iter_args[0] if len(iter_args) > 1 else self.visit(ast.Num(0))
+        ast_extent = iter_args[1] if len(iter_args) > 1 else iter_args[0]
+
+        # TODO(6clc): support sub region's local variable
+        llir_var = ir.Var(node.target.id)
+        llir_var_expr = ir.Expr(llir_var)
+        self.set_value(node.target.id, llir_var_expr)
+
+        llir_for_min = ir.Expr(ast_min)
```

Comment:
I guess LLIR For doesn't support `step`? Should we report error to users if user has `step` in for range?

Reply:
`step` has already been checked in the previous code, almost at line 109.

```python
        assert (
            len(iter_args) <= 2
        ), "CINN Low Level IR does not support setting the range step"
```